src/modules/dashboard/components/Dialogs/CreatePaymentDialog/CreatePaymentDialog.tsx

```diff
@@ -93,6 +98,42 @@ const CreatePaymentDialog = ({
     variables: { colonyAddress },
   });
 
+  const { data: colonyData } = useColonyFromNameQuery({
+    variables: { name: colonyName, address: colonyAddress },
+  });
+  const isWhiteListActive = colonyData?.processedColony?.isWhitelistActivated;
+
+  const { data } = useVerifiedUsersQuery({
+    variables: {
+      verifiedAddresses:
+        colonyData?.processedColony?.whitelistedAddresses || [],
+    },
+    fetchPolicy: 'network-only',
+  });
+
+  const storedVerifiedRecipients = useMemo(
+    () =>
+      (data?.verifiedUsers || []).map((user) => user?.profile.walletAddress),
+    [data],
+  );
+
+  const filteredVerifiedRecipients = isWhiteListActive
+    ? (colonyMembers?.subscribedUsers || []).filter((member) =>
+        storedVerifiedRecipients.some(
+          (el) => el.toLowerCase() === member.id.toLowerCase(),
+        ),
+      )
+    : colonyMembers?.subscribedUsers || [];
+
+  const getIsUnverifiedRecipient = (walletAddress) => {
+    if (!walletAddress) return false;
```

Comment:
I agree this could be confusing when viewed by fresh eyes. I have changed the func name to `showWarningForAddress`, then the return values make more sense.